res/controllers/Traktor-Kontrol-S3-hid-scripts.js

```diff
@@ -0,0 +1,2227 @@
+            this.hid.OutputPackets[packetName].send();
+        }
+    }
+};
+
+// Render wheel positions, channel VU meters, and master vu meters
+TraktorS3.Controller.prototype.guiTickHandler = function() {
+    this.batchingOutputs = true;
+    var gotUpdate = false;
+    gotUpdate |= this.Channels[this.Decks["deck1"].activeChannel].lightWheelPosition();
+    gotUpdate |= this.Channels[this.Decks["deck2"].activeChannel].lightWheelPosition();
+
+    for (var vu in this.masterVuMeter) {
+        if (this.masterVuMeter[vu].updated) {
+            this.vuMeterOutput(this.masterVuMeter[vu].value, "[Master]", vu, 8);
+            this.masterVuMeter[vu].updated = false;
+            gotUpdate = true;
+        }
+    }
+    for (var ch = 1; ch <= 4; ch++) {
+        var chan = this.Channels["[Channel" + ch + "]"];
+        if (chan.vuMeterUpdated) {
+            this.vuMeterOutput(chan.vuMeterValue, chan.group, "VuMeter", 14);
+            chan.vuMeterUpdated = false;
+            gotUpdate = true;
+        }
+    }
+
+    this.batchingOutputs = false;
+
+    if (gotUpdate) {
+        for (var packetName in this.hid.OutputPackets) {
+            this.hid.OutputPackets[packetName].send();
+        }
+    }
+};
+
+// A special packet sent to the controller switches between mic and line
+// input modes.  if lineMode is true, sets input to line. Otherwise, mic.
+TraktorS3.Controller.prototype.setInputLineMode = function(lineMode) {
+    var packet = Array();
+    packet.length = 33;
+    packet[0] = 0x20;
+    if (!lineMode) {
+        packet[1] = 0x08;
+    }
+    controller.send(packet, packet.length, 0xF4);
+};
+
+TraktorS3.messageCallback = function(_packet, data) {
+    for (var name in data) {
+        if (Object.prototype.hasOwnProperty.call(data, name)) {
+            TraktorS3.kontrol.hid.processButton(data[name]);
+        }
+    }
+};
+
+TraktorS3.incomingData = function(data, length) {
+    TraktorS3.kontrol.hid.parsePacket(data, length);
+};
+
+TraktorS3.debugLights = function() {
+    // Call this if you want to just send raw packets to the controller (good for figuring out what
+    // bytes do what).
+    var dataStrings = [
+        "00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "22 22 22 22  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "22 22 22 22  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "FF FF FF FF  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00 ",
+        "00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00",
+        "20 08 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00"
+    ];
```

Comment:
Why not use a normal JS Array?

Reply:
I do it like this because I am cutting and pasting the hex values from my USB sniffing program.  This way I can just paste in the values and run the code to see that the lights work the same was as in traktor.